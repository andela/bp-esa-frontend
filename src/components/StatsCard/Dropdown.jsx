import React, { PureComponent } from 'react';
import PropTypes from 'proptypes';
import ReactDOM from 'react-dom';
import './styles.scss';

class Dropdown extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      collapse: true,
      categoryTitle: 'today',
    };
  }

  showSection = () => {
    const { collapse } = this.state;
    const { collapseValue } = this.hideSection(collapse);
    this.setState({
      collapse: collapseValue,
    });
    document.addEventListener('click', this.closeStatDropdown);
  };

  hideSection = (collapse) => {
    let collapseValue;
    if (!collapse) {
      collapseValue = true;
    } else {
      collapseValue = false;
    }
    return { collapseValue };
  };

  closeStatDropdown = (event) => {
    const statsDropDown = ReactDOM.findDOMNode(this);
    if (event.target !== statsDropDown) {
      this.setState({
        collapse: true,
      });
      document.removeEventListener('click', this.closeStatDropdown);
    }
  };

  selectPeriod = (event) => {
    const { fetchStat } = this.props;
    const { innerHTML, id } = event.target;
    this.setState({
      collapse: true,
      categoryTitle: innerHTML,
    });
    fetchStat(id);
  }

  render() {
    const { collapse, categoryTitle } = this.state;
    return (
      <div className="stat-dropdown-container">
        Automations stats for
        <span className="stat-dropdown-caret" onClick={this.showSection}>
          {` ${categoryTitle}`}<img src="/drop-down.png" alt="down" />
        </span>
        {!collapse ? (
          <React.Fragment>
            <div className="stat-dropdown-content">
              <div className="stat-dropdown-item" id="days" onClick={this.selectPeriod}>
                today
              </div>
              <div className="stat-dropdown-item" id="weeks" onClick={this.selectPeriod}>
                this week
              </div>
              <div className="stat-dropdown-item" id="months" onClick={this.selectPeriod}>
                this month
              </div>
              <div className="stat-dropdown-item" id="years" onClick={this.selectPeriod}>
                this year
              </div>
            </div>
          </React.Fragment>
        ) : null}
      </div>
    );
  }
}

Dropdown.propTypes = {
  fetchStat: PropTypes.func.isRequired,
};

export default Dropdown;
