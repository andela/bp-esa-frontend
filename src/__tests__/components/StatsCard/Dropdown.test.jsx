import React from 'react';
import { mount } from 'enzyme';
import Dropdown from '../../../components/StatsCard/Dropdown';


const fetchStat = jest.fn();
let wrapper;

describe('Dropdown', () => {
  beforeEach(() => {
    wrapper = mount(<Dropdown fetchStat={fetchStat} />);
  });
  it('should render as expected', () => {
    expect(wrapper.instance()).toMatchSnapshot();
  });

  it('should show the dropdown when showSection is called with the dropdown hidden', () => {
    wrapper.setState({
      collapse: true,
    });
    wrapper.instance().showSection();
    expect(wrapper.state().collapse).toEqual(false);
  });

  it('should hide the dropdown when showSection is called with the dropdown showing', () => {
    wrapper.setState({
      collapse: false,
    });
    wrapper.instance().showSection();
    expect(wrapper.state().collapse).toEqual(true);
  });

  it('should set state and call the fetchStat prop with form input id passed as a parameter', () => {
    wrapper.setState({
      collapse: false,
      categoryTitle: 'this week',
    });
    const event = {
      target: {
        id: 'days',
        innerHTML: 'today',
      },
    };
    wrapper.instance().selectPeriod(event);
    expect(wrapper.state().collapse).toEqual(true);
    expect(wrapper.state().categoryTitle).toEqual('today');
    expect(fetchStat).toBeCalledWith(event.target.id);
  });

  it('should hide the dropdown when a user clicks outside the dropdown', () => {
    wrapper.setState({
      collapse: false,
    });
    const event = {
      target: {
        notStatsDropDown: true,
      },
    };
    wrapper.instance().closeStatDropdown(event);
    expect(wrapper.state().collapse).toEqual(true);
  });
});
