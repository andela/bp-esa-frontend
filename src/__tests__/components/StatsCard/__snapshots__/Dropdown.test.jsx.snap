// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render as expected 1`] = `
Dropdown {
  "_reactInternalFiber": FiberNode {
    "_debugHookTypes": null,
    "_debugID": 123,
    "_debugIsCurrentlyTiming": false,
    "_debugNeedsRemount": false,
    "_debugOwner": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 121,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": [Circular],
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 1,
      "elementType": [Function],
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "Component": [Function],
        "context": null,
        "props": Object {
          "fetchStat": [MockFunction],
        },
        "wrappingComponentProps": null,
      },
      "memoizedState": Object {
        "context": null,
        "mount": true,
        "props": Object {
          "fetchStat": [MockFunction],
        },
        "wrappingComponentProps": null,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "Component": [Function],
        "context": null,
        "props": Object {
          "fetchStat": [MockFunction],
        },
        "wrappingComponentProps": null,
      },
      "ref": null,
      "return": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 118,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 118,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": null,
          "_debugSource": null,
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": [Circular],
          "child": null,
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": null,
          "expirationTime": 1073741823,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": null,
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": null,
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": FiberRootNode {
            "callbackExpirationTime": 0,
            "callbackNode": null,
            "containerInfo": <div>
              <div
                class="stat-dropdown-container"
              >
                Automations stats for
                <span
                  class="stat-dropdown-caret"
                >
                   today
                  <img
                    alt="down"
                    src="/drop-down.png"
                  />
                </span>
              </div>
            </div>,
            "context": Object {},
            "current": [Circular],
            "finishedExpirationTime": 0,
            "finishedWork": null,
            "firstBatch": null,
            "firstPendingTime": 0,
            "hydrate": false,
            "interactionThreadID": 16,
            "lastPendingTime": 0,
            "memoizedInteractions": Set {},
            "pendingChildren": null,
            "pendingContext": null,
            "pendingInteractionMap": Map {},
            "pingCache": null,
            "pingTime": 0,
            "tag": 0,
            "timeoutHandle": -1,
          },
          "tag": 3,
          "treeBaseDuration": 0,
          "type": null,
          "updateQueue": Object {
            "baseState": null,
            "firstCapturedEffect": null,
            "firstCapturedUpdate": null,
            "firstEffect": null,
            "firstUpdate": Object {
              "callback": null,
              "expirationTime": 1073741823,
              "next": null,
              "nextEffect": null,
              "payload": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "fetchStat": [MockFunction],
                    }
                  }
                  wrappingComponentProps={null}
                />,
              },
              "priority": 97,
              "suspenseConfig": null,
              "tag": 0,
            },
            "lastCapturedEffect": null,
            "lastCapturedUpdate": null,
            "lastEffect": null,
            "lastUpdate": Object {
              "callback": null,
              "expirationTime": 1073741823,
              "next": null,
              "nextEffect": null,
              "payload": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "fetchStat": [MockFunction],
                    }
                  }
                  wrappingComponentProps={null}
                />,
              },
              "priority": 97,
              "suspenseConfig": null,
              "tag": 0,
            },
          },
        },
        "child": [Circular],
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 0,
        "firstEffect": [Circular],
        "index": 0,
        "key": null,
        "lastEffect": [Circular],
        "memoizedProps": null,
        "memoizedState": Object {
          "element": <WrapperComponent
            Component={[Function]}
            context={null}
            props={
              Object {
                "fetchStat": [MockFunction],
              }
            }
            wrappingComponentProps={null}
          />,
        },
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": FiberRootNode {
          "callbackExpirationTime": 0,
          "callbackNode": null,
          "containerInfo": <div>
            <div
              class="stat-dropdown-container"
            >
              Automations stats for
              <span
                class="stat-dropdown-caret"
              >
                 today
                <img
                  alt="down"
                  src="/drop-down.png"
                />
              </span>
            </div>
          </div>,
          "context": Object {},
          "current": [Circular],
          "finishedExpirationTime": 0,
          "finishedWork": null,
          "firstBatch": null,
          "firstPendingTime": 0,
          "hydrate": false,
          "interactionThreadID": 16,
          "lastPendingTime": 0,
          "memoizedInteractions": Set {},
          "pendingChildren": null,
          "pendingContext": null,
          "pendingInteractionMap": Map {},
          "pingCache": null,
          "pingTime": 0,
          "tag": 0,
          "timeoutHandle": -1,
        },
        "tag": 3,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": Object {
          "baseState": Object {
            "element": <WrapperComponent
              Component={[Function]}
              context={null}
              props={
                Object {
                  "fetchStat": [MockFunction],
                }
              }
              wrappingComponentProps={null}
            />,
          },
          "firstCapturedEffect": null,
          "firstCapturedUpdate": null,
          "firstEffect": null,
          "firstUpdate": null,
          "lastCapturedEffect": null,
          "lastCapturedUpdate": null,
          "lastEffect": null,
          "lastUpdate": null,
        },
      },
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": WrapperComponent {
        "_reactInternalFiber": [Circular],
        "_reactInternalInstance": Object {},
        "context": Object {},
        "props": Object {
          "Component": [Function],
          "context": null,
          "props": Object {
            "fetchStat": [MockFunction],
          },
          "wrappingComponentProps": null,
        },
        "refs": Object {},
        "rootFinderInstance": null,
        "state": Object {
          "context": null,
          "mount": true,
          "props": Object {
            "fetchStat": [MockFunction],
          },
          "wrappingComponentProps": null,
        },
        "updater": Object {
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
        },
      },
      "tag": 1,
      "treeBaseDuration": 0,
      "type": [Function],
      "updateQueue": null,
    },
    "_debugSource": null,
    "actualDuration": 0,
    "actualStartTime": -1,
    "alternate": null,
    "child": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 125,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "fileName": "/Users/arnold/Andela-Project/bp-esa-frontend/src/components/StatsCard/Dropdown.jsx",
        "lineNumber": 57,
      },
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 127,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": null,
        "child": null,
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": "Automations stats for",
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": "Automations stats for",
        "ref": null,
        "return": [Circular],
        "selfBaseDuration": 0,
        "sibling": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 128,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": [Circular],
          "_debugSource": Object {
            "fileName": "/Users/arnold/Andela-Project/bp-esa-frontend/src/components/StatsCard/Dropdown.jsx",
            "lineNumber": 59,
          },
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": null,
          "child": FiberNode {
            "_debugHookTypes": null,
            "_debugID": 131,
            "_debugIsCurrentlyTiming": false,
            "_debugNeedsRemount": false,
            "_debugOwner": null,
            "_debugSource": null,
            "actualDuration": 0,
            "actualStartTime": -1,
            "alternate": null,
            "child": null,
            "childExpirationTime": 0,
            "dependencies": null,
            "effectTag": 0,
            "elementType": null,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": " today",
            "memoizedState": null,
            "mode": 0,
            "nextEffect": null,
            "pendingProps": " today",
            "ref": null,
            "return": [Circular],
            "selfBaseDuration": 0,
            "sibling": FiberNode {
              "_debugHookTypes": null,
              "_debugID": 132,
              "_debugIsCurrentlyTiming": false,
              "_debugNeedsRemount": false,
              "_debugOwner": [Circular],
              "_debugSource": Object {
                "fileName": "/Users/arnold/Andela-Project/bp-esa-frontend/src/components/StatsCard/Dropdown.jsx",
                "lineNumber": 60,
              },
              "actualDuration": 0,
              "actualStartTime": -1,
              "alternate": null,
              "child": null,
              "childExpirationTime": 0,
              "dependencies": null,
              "effectTag": 0,
              "elementType": "img",
              "expirationTime": 0,
              "firstEffect": null,
              "index": 1,
              "key": null,
              "lastEffect": null,
              "memoizedProps": Object {
                "alt": "down",
                "src": "/drop-down.png",
              },
              "memoizedState": null,
              "mode": 0,
              "nextEffect": null,
              "pendingProps": Object {
                "alt": "down",
                "src": "/drop-down.png",
              },
              "ref": null,
              "return": [Circular],
              "selfBaseDuration": 0,
              "sibling": null,
              "stateNode": <img
                alt="down"
                src="/drop-down.png"
              />,
              "tag": 5,
              "treeBaseDuration": 0,
              "type": "img",
              "updateQueue": null,
            },
            "stateNode":  today,
            "tag": 6,
            "treeBaseDuration": 0,
            "type": null,
            "updateQueue": null,
          },
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": "span",
          "expirationTime": 0,
          "firstEffect": null,
          "index": 1,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "children": Array [
              " today",
              <img
                alt="down"
                src="/drop-down.png"
              />,
            ],
            "className": "stat-dropdown-caret",
            "onClick": [Function],
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "children": Array [
              " today",
              <img
                alt="down"
                src="/drop-down.png"
              />,
            ],
            "className": "stat-dropdown-caret",
            "onClick": [Function],
          },
          "ref": null,
          "return": [Circular],
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": <span
            class="stat-dropdown-caret"
          >
             today
            <img
              alt="down"
              src="/drop-down.png"
            />
          </span>,
          "tag": 5,
          "treeBaseDuration": 0,
          "type": "span",
          "updateQueue": null,
        },
        "stateNode": Automations stats for,
        "tag": 6,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": null,
      },
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 0,
      "elementType": "div",
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": Array [
          "Automations stats for",
          <span
            className="stat-dropdown-caret"
            onClick={[Function]}
          >
             today
            <img
              alt="down"
              src="/drop-down.png"
            />
          </span>,
          null,
        ],
        "className": "stat-dropdown-container",
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": Array [
          "Automations stats for",
          <span
            className="stat-dropdown-caret"
            onClick={[Function]}
          >
             today
            <img
              alt="down"
              src="/drop-down.png"
            />
          </span>,
          null,
        ],
        "className": "stat-dropdown-container",
      },
      "ref": null,
      "return": [Circular],
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": <div
        class="stat-dropdown-container"
      >
        Automations stats for
        <span
          class="stat-dropdown-caret"
        >
           today
          <img
            alt="down"
            src="/drop-down.png"
          />
        </span>
      </div>,
      "tag": 5,
      "treeBaseDuration": 0,
      "type": "div",
      "updateQueue": null,
    },
    "childExpirationTime": 0,
    "dependencies": null,
    "effectTag": 1,
    "elementType": [Function],
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {
      "fetchStat": [MockFunction],
    },
    "memoizedState": Object {
      "categoryTitle": "today",
      "collapse": true,
    },
    "mode": 0,
    "nextEffect": null,
    "pendingProps": Object {
      "fetchStat": [MockFunction],
    },
    "ref": null,
    "return": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 121,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": [Circular],
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 1,
      "elementType": [Function],
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "Component": [Function],
        "context": null,
        "props": Object {
          "fetchStat": [MockFunction],
        },
        "wrappingComponentProps": null,
      },
      "memoizedState": Object {
        "context": null,
        "mount": true,
        "props": Object {
          "fetchStat": [MockFunction],
        },
        "wrappingComponentProps": null,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "Component": [Function],
        "context": null,
        "props": Object {
          "fetchStat": [MockFunction],
        },
        "wrappingComponentProps": null,
      },
      "ref": null,
      "return": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 118,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 118,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": null,
          "_debugSource": null,
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": [Circular],
          "child": null,
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": null,
          "expirationTime": 1073741823,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": null,
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": null,
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": FiberRootNode {
            "callbackExpirationTime": 0,
            "callbackNode": null,
            "containerInfo": <div>
              <div
                class="stat-dropdown-container"
              >
                Automations stats for
                <span
                  class="stat-dropdown-caret"
                >
                   today
                  <img
                    alt="down"
                    src="/drop-down.png"
                  />
                </span>
              </div>
            </div>,
            "context": Object {},
            "current": [Circular],
            "finishedExpirationTime": 0,
            "finishedWork": null,
            "firstBatch": null,
            "firstPendingTime": 0,
            "hydrate": false,
            "interactionThreadID": 16,
            "lastPendingTime": 0,
            "memoizedInteractions": Set {},
            "pendingChildren": null,
            "pendingContext": null,
            "pendingInteractionMap": Map {},
            "pingCache": null,
            "pingTime": 0,
            "tag": 0,
            "timeoutHandle": -1,
          },
          "tag": 3,
          "treeBaseDuration": 0,
          "type": null,
          "updateQueue": Object {
            "baseState": null,
            "firstCapturedEffect": null,
            "firstCapturedUpdate": null,
            "firstEffect": null,
            "firstUpdate": Object {
              "callback": null,
              "expirationTime": 1073741823,
              "next": null,
              "nextEffect": null,
              "payload": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "fetchStat": [MockFunction],
                    }
                  }
                  wrappingComponentProps={null}
                />,
              },
              "priority": 97,
              "suspenseConfig": null,
              "tag": 0,
            },
            "lastCapturedEffect": null,
            "lastCapturedUpdate": null,
            "lastEffect": null,
            "lastUpdate": Object {
              "callback": null,
              "expirationTime": 1073741823,
              "next": null,
              "nextEffect": null,
              "payload": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "fetchStat": [MockFunction],
                    }
                  }
                  wrappingComponentProps={null}
                />,
              },
              "priority": 97,
              "suspenseConfig": null,
              "tag": 0,
            },
          },
        },
        "child": [Circular],
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 0,
        "firstEffect": [Circular],
        "index": 0,
        "key": null,
        "lastEffect": [Circular],
        "memoizedProps": null,
        "memoizedState": Object {
          "element": <WrapperComponent
            Component={[Function]}
            context={null}
            props={
              Object {
                "fetchStat": [MockFunction],
              }
            }
            wrappingComponentProps={null}
          />,
        },
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": FiberRootNode {
          "callbackExpirationTime": 0,
          "callbackNode": null,
          "containerInfo": <div>
            <div
              class="stat-dropdown-container"
            >
              Automations stats for
              <span
                class="stat-dropdown-caret"
              >
                 today
                <img
                  alt="down"
                  src="/drop-down.png"
                />
              </span>
            </div>
          </div>,
          "context": Object {},
          "current": [Circular],
          "finishedExpirationTime": 0,
          "finishedWork": null,
          "firstBatch": null,
          "firstPendingTime": 0,
          "hydrate": false,
          "interactionThreadID": 16,
          "lastPendingTime": 0,
          "memoizedInteractions": Set {},
          "pendingChildren": null,
          "pendingContext": null,
          "pendingInteractionMap": Map {},
          "pingCache": null,
          "pingTime": 0,
          "tag": 0,
          "timeoutHandle": -1,
        },
        "tag": 3,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": Object {
          "baseState": Object {
            "element": <WrapperComponent
              Component={[Function]}
              context={null}
              props={
                Object {
                  "fetchStat": [MockFunction],
                }
              }
              wrappingComponentProps={null}
            />,
          },
          "firstCapturedEffect": null,
          "firstCapturedUpdate": null,
          "firstEffect": null,
          "firstUpdate": null,
          "lastCapturedEffect": null,
          "lastCapturedUpdate": null,
          "lastEffect": null,
          "lastUpdate": null,
        },
      },
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": WrapperComponent {
        "_reactInternalFiber": [Circular],
        "_reactInternalInstance": Object {},
        "context": Object {},
        "props": Object {
          "Component": [Function],
          "context": null,
          "props": Object {
            "fetchStat": [MockFunction],
          },
          "wrappingComponentProps": null,
        },
        "refs": Object {},
        "rootFinderInstance": null,
        "state": Object {
          "context": null,
          "mount": true,
          "props": Object {
            "fetchStat": [MockFunction],
          },
          "wrappingComponentProps": null,
        },
        "updater": Object {
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
        },
      },
      "tag": 1,
      "treeBaseDuration": 0,
      "type": [Function],
      "updateQueue": null,
    },
    "selfBaseDuration": 0,
    "sibling": null,
    "stateNode": [Circular],
    "tag": 1,
    "treeBaseDuration": 0,
    "type": [Function],
    "updateQueue": null,
  },
  "_reactInternalInstance": Object {},
  "closeStatDropdown": [Function],
  "context": Object {},
  "hideSection": [Function],
  "props": Object {
    "fetchStat": [MockFunction],
  },
  "refs": Object {},
  "selectPeriod": [Function],
  "showSection": [Function],
  "state": Object {
    "categoryTitle": "today",
    "collapse": true,
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
